tools/awslint/lib/rules/construct.ts

```diff
@@ -227,4 +207,32 @@ constructLinter.add({
       e.assert(!property.name.toLowerCase().endsWith('arn'), `${e.ctx.propsFqn}.${property.name}`);
     }
   }
+});
+
+constructLinter.add({
+  code: 'props-no-tokens',
+  message: 'props should not use the "Token" type',
+  eval: e => {
+    if (!e.ctx.propsType) { return; }
+    if (!e.ctx.hasPropsArgument) { return; }
+
+    // this rule only applies to L2 constructs
+    if (e.ctx.classType.name.startsWith('Cfn')) { return; }
+
+    for (const property of e.ctx.propsType.allProperties) {
+      const typeRef = property.type;
+      let fqn = typeRef.fqn;
+
+      if (typeRef.arrayOfType) {
+        fqn = typeRef.arrayOfType.fqn;
+      } else if (typeRef.mapOfType) {
+        fqn = typeRef.mapOfType.fqn;
+      }
+
+      const found = (fqn && e.ctx.sys.tryFindFqn(fqn));
+      if(found) {
+        e.assert(!(fqn === e.ctx.core.tokenClass.fqn), `${e.ctx.propsFqn}.${property.name}`);
```

Comment:
The assert message should probably include a descriptive text in addition to just the property name.

Reply:
The property name is just the scope of the rule. The text describing the violation comes from the message field. 

eg:

```
error: [awslint:props-no-tokens:@aws-cdk/aws-resource.ResourceProps.badParameter] props should not use the "Token" type 
```

Am I missing something?

Reply:
Nope, you are correct, just misunderstood the purpose of the argument.